client-app/core/composables/useBreadcrumbs.ts

```diff
@@ -0,0 +1,18 @@
+import { useI18n } from "vue-i18n";
+import { MaybeRef } from "@vueuse/core";
+import { computed, Ref, unref } from "vue";
+
+interface UseBreadcrumbs {
+  breadcrumbs: Ref<IBreadcrumb[]>;
+}
+
+export function useBreadcrumbs(breadcrumbs: MaybeRef<IBreadcrumb[]>): UseBreadcrumbs {
+  const { t } = useI18n();
+
+  return {
+    breadcrumbs: computed(() => {
```

Comment:
Wouldn't it be better to return the breadcrumbs right away?

Reply:
[No](https://vuejs.org/guide/reusability/composables.html#return-values)

Reply:
> The recommended convention is for composables to always return a plain, non-reactive object containing **multiple** refs.

> containing **multiple** ref

on that same page below:

> [VueUse](https://vueuse.org/): an ever-growing collection of Vue composables. The source code is also a great learning resource.

https://vuejs.org/guide/reusability/composables.html#further-reading

VueUse has clear examples. For example: https://vueuse.org/core/useActiveElement/#usage
![image](https://user-images.githubusercontent.com/37446577/216035899-770f40ee-2377-4f2c-8b3f-39a4ee450f7d.png)

Reply:
Refactored